src/Workspaces/Core/Portable/Editing/DeclarationModifiers.cs

```diff
@@ -203,6 +211,7 @@ private enum Modifiers
             Volatile    = 1 << 14,
             Extern      = 1 << 15,
             Required    = 1 << 16,
+            File        = 1 << 17,
```

Comment:
Since you added this API, consider fixing up:

Consider also updating

https://github.com/dotnet/roslyn/blob/5510ef93015c7123d20c2e336c94549bd2116a76/src/Workspaces/CSharp/Portable/CodeGeneration/CSharpSyntaxGenerator.cs#L1491-L1492

https://github.com/dotnet/roslyn/blob/5510ef93015c7123d20c2e336c94549bd2116a76/src/Workspaces/CSharp/Portable/CodeGeneration/CSharpSyntaxGenerator.cs#L1444-L1466

Reply:
It actually wasn't clear to me whether 'file' should be added to this. I assume if the IDE has need to generate syntax for a file type, it would use this. But I don't know if the use case exists. It would be helpful to know whether I should delete and just stub out a `bool IsFIle => false;` implementation.

Reply:
@RikkiGibson This is a public API. So it's very necessary that 'file' is added here (not necessarily in this PR)

A common SyntaxGenerator pattern is `generator.WithModifiers(node, generator.GetModifiers(node).WithX(true))`

If the node with a class declaration for `file class C { }`, and we called `WithSealed(true)` (for example), I expect the resulting node will drop `file` if the change is not made.

Reply:
@Youssef1313  is correct.  thanks! :)